packages/framework/src/errors/import.errors.ts

```diff
@@ -0,0 +1,16 @@
+import { ErrorCodeEnum, HttpStatusEnum } from '../constants';
+import { ServerError } from './base.errors';
+
+export class MissingDependencyError extends ServerError {
+  statusCode = HttpStatusEnum.INTERNAL_SERVER_ERROR;
+  code = ErrorCodeEnum.MISSING_DEPENDENCY_ERROR;
+
+  constructor(usageReason: string, missingDependencies: string[]) {
+    const pronoun = missingDependencies.length === 1 ? 'it' : 'them';
+    super(
+      `Tried to use a ${usageReason} in @novu/framework without ${missingDependencies.join(
+        ', '
+      )} installed. Please install ${pronoun} by running \`npm install ${missingDependencies.join(' ')}\`.`
+    );
+  }
+}
```

Comment:
We now have a generic error for all missing dependency errors. This covers only validators right now in the implementation, but can be extended to other plugin domains.

Reply:
100%. I will address documentation for this case when writing docs for Class Validators.